modules/phase_field/src/ics/IsolatedBoundingBoxIC.C

```diff
@@ -0,0 +1,133 @@
+
+  params.addClassDescription("Allows setting the initial condition of the value of a field inside "
+                             "multiple isolated bounding boxes.");
+  return params;
+}
+
+IsolatedBoundingBoxIC::IsolatedBoundingBoxIC(const InputParameters & parameters)
+  : InitialCondition(parameters),
+    _c1(getParam<std::vector<Point>>("smaller_coordinate_corners")),
+    _c2(getParam<std::vector<Point>>("larger_coordinate_corners")),
+    _nbox(_c1.size()),
+    _int_width(getParam<Real>("int_width")),
+    _dim(_fe_problem.mesh().dimension()),
+    _inside(sizeVector_def_isolated::sizeVector(getParam<std::vector<Real>>("inside"), _nbox)),
+    _outside(getParam<Real>("outside"))
+{
+  // make sure inputs are the same length
+  if (_c2.size() != _nbox || _inside.size() != _nbox)
+    mooseError("vector inputs must all be the same size");
+}
+
+Real
+IsolatedBoundingBoxIC::value(const Point & p)
+{
+  Real value = _outside; // if p is not inside the largest box, assign the last element of inside
+                         // which is 'outside' to value
+
+  if (_int_width < 0.0)
+    mooseError("'int_width' should be non-negative");
+
+  if (_int_width == 0.0)
+  {
+    for (unsigned int b = 0; b < _nbox; ++b)
+    {
+      if (p(0) >= _c1[b](0) && p(0) <= _c2[b](0))
+        if (_dim <= 1 || (_c1[b](1) < _c2[b](1) && p(1) >= _c1[b](1) && p(1) <= _c2[b](1)))
+          if (_dim <= 2 || (_c1[b](2) < _c2[b](2) && p(2) >= _c1[b](2) && p(2) <= _c2[b](2)))
+          {
+            value = _inside[b];
+            break;
+          }
+    }
+  }
+
+  else
+  {
+    for (unsigned int b = 0; b < _nbox; ++b)
+    {
+      if (p(0) >= _c1[b](0) - _int_width && p(0) <= _c2[b](0) + _int_width)
+        if (_dim <= 1 || (p(1) >= _c1[b](1) - _int_width && p(1) <= _c2[b](1) + _int_width))
+          if (_dim <= 2 || (p(2) >= _c1[b](2) - _int_width && p(2) <= _c2[b](2) + _int_width))
+          {
+            for (unsigned int n = b + 1; n < _nbox; ++n)
+            {
+              if (p(0) >= _c1[n](0) - _int_width && p(0) <= _c2[n](0) + _int_width)
+                if (_dim <= 1 || (p(1) >= _c1[n](1) - _int_width && p(1) <= _c2[n](1) + _int_width))
+                  if (_dim <= 2 ||
+                      (p(2) >= _c1[n](2) - _int_width && p(2) <= _c2[n](2) + _int_width))
+                  {
+                    mooseError("Partially overlapping boxes are not allowed. Note that this "
+                               "includes the overlapping diffused interfaces. For nested boxes, "
+                               "use NestedBoundingBoxIC.C");
+                  }
+            }
+
+            Real f_in = 1.0;
+            for (unsigned int i = 0; i < LIBMESH_DIM; ++i)
+              if (_c1[b](i) != _c2[b](i))
+                f_in *= 0.5 * (std::tanh(2.0 * 3.1415926 * (p(i) - _c1[b](i)) / _int_width) -
+                               std::tanh(2.0 * 3.1415926 * (p(i) - _c2[b](i)) / _int_width));
```

Comment:
There is a libMesh::PI constant you can use. You'll definitely want that to make sure you get the maximum precision for PI when using a double.